server/app/views/admin/programs/ProgramStatusesView.java

```diff
@@ -47,31 +48,45 @@ public final class ProgramStatusesView extends BaseHtmlView {
   public static final String DELETE_STATUS_TEXT_NAME = "deleteStatusText";
 
   private final AdminLayout layout;
+  private final FormFactory formFactory;
   private final MessagesApi messagesApi;
 
   @Inject
-  public ProgramStatusesView(AdminLayoutFactory layoutFactory, MessagesApi messagesApi) {
+  public ProgramStatusesView(
+      AdminLayoutFactory layoutFactory, FormFactory formFactory, MessagesApi messagesApi) {
     this.layout = checkNotNull(layoutFactory).getLayout(NavPage.PROGRAMS);
+    this.formFactory = formFactory;
     this.messagesApi = checkNotNull(messagesApi);
   }
 
+  /**
+   * Renders a list of program statuses along with modal create / edit / delete forms.
+   *
+   * @param request The associated request.
+   * @param program The program to render program statuses for.
+   * @param maybeStatusForm Set if the form is being rendered in response to an attempt to create /
+   *     edit a program status. Note that while the view itself may render multiple program
+   *     statuses, the provided form will correspond to creating / editing a single program status.
+   */
   public Content render(
       Http.Request request,
       ProgramDefinition program,
-      Optional<Form<ProgramStatusesEditForm>> maybeEditForm) {
-    Modal createStatusModal =
-        makeStatusEditModal(
-            request,
-            program,
-            Optional.empty(),
-            formForCurrentStatus(Optional.empty(), maybeEditForm));
+      Optional<Form<ProgramStatusesForm>> maybeStatusForm) {
+    final Form<ProgramStatusesForm> createStatusForm;
+    if (isFormForCurrentStatus(Optional.empty(), maybeStatusForm)) {
```

Comment:
isFormForCurrentStatus doesn't need to accept an Optional Status as it will always return false, and as such doesn't need to be called in this situation. 

That said, this if will never be true, so what's up with the if/else?

Reply:
It does in the case that we're rendering the creation form. There isn't a configured status (e.g. Optional.empty()) and there is a form that has the configuredStatusText field set to empty.

> That said, this if will never be true, so what's up with the if/else?
It will in the situation described above.

Reply:
I thin it'd be a lot cleaner to have a method that knowingly determines if it's a create action rather than encoding create as empty/"" value, and then isFormForCurrentStatus does a check for an actual value.

boolean isCreationForm(form) {
  return maybeEditForm.get().value().get().getConfiguredStatusText().isEmpty();
}

Reply:
I have a slight preference for a single method (so that the matching logic is consolidated to a single method). In my opinion, that potentially reduces the maintenance burden if we decide to change semantics on what form fields mean. That said, it's nebulous and I don't feel strongly enough to keep pushing for it. Introduced the method you described.